lib/payment_processor/braintree/webhook_handler.rb

```diff
@@ -77,6 +77,25 @@ def handle_subscription_cancelled
 
       def handle_subscription_charge(status)
         return unless subscription
+        update_subscription(status)
+        create_subscription_charge(status)
+        true
+      end
+
+      def subscription_amount
+        # The subscription hook contains an array of transactions with the latest one first.
+        @subscription_amount ||= notification.subscription.transactions.first&.amount || 0
```

Comment:
Nah, amount would be 0 only if the first subscription charge was unsuccessful. It's a rare scenario but we've run into it a few times. In this case we don't update the subscription.